app/src/main/java/com/pixlee/pixleeandroidsdk/ui/AnalyticsFragment.java

```diff
@@ -124,13 +133,21 @@ private void setClickListeners() {
         bt_open_widget.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                showMessage("openWidget(..)");
-
+                showMessage("openedWidget(..)");
                 album.openedWidget(PXLWidgetType.photowall);
                 // Alternative: album.openedWidget("<Customized name>");
             }
         });
 
+        bt_widget_visible.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                showMessage("widgetVisible(..)");
+                album.widgetVisible(PXLWidgetType.photowall);
```

Comment:
what does this onClick stuff do?

Reply:
onClick is fired, when the button that is listening, is tapped by the user. This can be the same as **onclick** of HTML.
```
<input type="checkbox" id="myCheck" onmouseover="myFunction()" onclick="alert('click event occured')">
```

AnalyticsFragment.java presents a UI having buttons. **onclick** line 144 is listening to the second button, **Widget Visible**, in the screenshot below.
<img src="https://github.com/pixlee/android-sdk/blob/feature/add_widget_visible/doc/img/demo_4_analytics.jpg?raw=true" width="50%">

Reply:
ah cool!